send2cgeo.user.js

```diff
@@ -25,27 +24,163 @@
+    + '        },'
+    + '        100'
+    + '    );'
+    + '};';
+
+document.getElementsByTagName('head')[0].appendChild(s2geomultiScript);
+
+var start = function(c) {
+    quitOnAdFrames()
+        .then(function() {
+            return jqueryInit(c);
+        })
+        .done(function() {
+            s2cgGCMain();
+        });
+};
+
+// This solves the problems with jquery
+var quitOnAdFrames = function(c) {
+    var quitOnAdFramesDeref = new jQuery.Deferred();
+    if(window.name) {
+        if (window.name.substring(0, 18) !== 'google_ads_iframe_') {
+            quitOnAdFramesDeref.resolve();
+        }else {
+            quitOnAdFramesDeref.reject();
+        }
+    }else {
+        quitOnAdFramesDeref.resolve();
+    }
+    return quitOnAdFramesDeref.promise();
+};
 
-s.type = 'text/javascript';
-s.textContent = '(' + function() {
-    // function that handles the actual sending //////////////////////////////////
+var jqueryInit = function(c) {
+    if (typeof c.$ === "undefined") {
+        c.$ = c.$ || unsafeWindow.$ || window.$ || null;
+    }
+    if (typeof c.jQuery === "undefined") {
+        c.jQuery = c.jQuery || unsafeWindow.jQuery || window.jQuery || null;
+    }
+    var jqueryInitDeref = new jQuery.Deferred();
+    jqueryInitDeref.resolve();
+    return jqueryInitDeref.promise();
+};
+
+// On the new searchmap the button doesn't work. This function will be insert a button that works.
+function addSend2cgeoButton(anchorElem, insertMethod='append', GCCode, withText=false, enclosingTag='', css='') {
+    // Break when a button with the GCCode alrady exist
+    if (document.getElementById('s2cg-' + GCCode)) {
+        return;
+    }
+    var html = enclosingTag;
+    html += '<a style="cursor:pointer;" id="s2cg-' + GCCode + '" title="Send to c:geo" class="s2cg">';
+    html += '<img src="https://send2.cgeo.org/send2cgeo.png" />';
+    if (withText) {
+        html += '<span>Send to c:geo</span>';
+    }
+    html += '</a>' + enclosingTag.replace('<', '</');
+    switch (insertMethod) {
+        case 'after': anchorElem.after(html);
+        case 'before': anchorElem.before(html);
+        case 'append': anchorElem.append(html);
+    }
 
-    window.s2geo = function(code) {
+    if (css != '') {
+        var style = document.createElement('style');
+        style.innerHTML = 's2sg{} ' + css;
+        style.type = 'text/css';
+        $('head')[0].appendChild(style);
+    }
+        
+    // function that handles the actual sending
+    $('#s2cg-' + GCCode).bind('click', function(){
         // show the box and the "please wait" text
         $("#send2cgeo, #send2cgeo div").fadeIn();
         // hide iframe for now and wait for page to be loaded
         $("#send2cgeo iframe")
             .hide()
             .off('load')
-            .attr('src', 'https://send2.cgeo.org/add.html?cache=' + code)
+            .attr('src', 'https://send2.cgeo.org/add.html?cache=' + GCCode)
```

Comment:
Please refaktor: only use "code" or "GCCode" in the script for the same meaning.